### ISSUE TITLE
[FEATURE] Allow use of env variables inside in stacks too

### ISSUE THREAD

**Issue by vikas027** [terramate-io/terramate#1384]
**Is your feature request related to a problem? Please describe.**

I am working on Azure and have a multi-subscription architecture. If I am sitting on root and I run `terramate run terraform init` command, it traverses down to each directory and executes the command.

For example, if my stack structure is like below, I want to use environment variables (`ARM_SUBSCRIPTION_ID`, `ARM_TENANT_ID`, `ARM_CLIENT_ID` and `ARM_CLIENT_SECRET`) specific to an environment.

```bash
envs/dev/terraform_init/stack.tm.hcl
envs/prod/terraform_init/stack.tm.hcl
```

**Describe the solution you'd like**

Any way of picking environment variables would be good, maybe something like this

```hcl
stack {
  name        = "terraform_init"
  description = "Initialize Terraform Backend"
  id          = "xxxx-yyy"

  tags = [
    "dev",
  ]

  env_vars = [
    ARM_CLIENT_SECRET = "${env.DEV_ARM_CLIENT_SECRET}"
  ]
}
```

**Describe alternatives you've considered**

As of now, I am using these commands to individually run each of the stacks after setting up AZ subscription

```bash
source dev.env
terramate run -C envs/dev terraform init

source prod.env
terramate run -C envs/prod terraform init
```

**Comment by vikas027** [comment]
Hey @soerenmartius ,

Yes, it would be good to inherit an environment variable and inherit it in all stacks. This will be helpful in my situation where I want to keep the code DRY in a multi-account / region setup.

**Comment by mariux** [comment]
Hi @vikas027,

So we planned the following change in defining the run time environment:

We will allow to set `terramate.config.run.env` variables at any level of the hierarchy. Inheritance will be similar to globals:

- any variable defined on a higher level will be inherited to the full subtree including nested stacks.
- any variable redefined on the same level will conflict
- any variable redefined on lower levels will overwrite previous definitions
- all values will be only evaluated on stack level.
- any `null` value will be treated as `unset` and not variable will be exported
- to unset a variable on a lower level you can assign `unset` value or `null`

Before we start, could you confirm if this would solve your problem?

**Comment by g13013** [comment]
@mariux the `env` block would be under `terramate.config.run.env` or `stack.run.env` ? the later makes more sense to me as it clearly implies that it's scoped to the stack.

**Comment by soerenmartius** [comment]
> @mariux the `env` block would be under `terramate.config.run.env` or `stack.run.env` ? the later makes more sense to me as it clearly implies that it's scoped to the stack.

Using `terramate.config.run.env` is already possible in stacks and any other point in your hierarchy since [`v0.8.0`](https://github.com/terramate-io/terramate/releases/tag/v0.8.0). It will be inherited for nested stacks though. 

Perhaps we could think about adding `stack.run.env` to define environment variables for one specific stack only. I'll discuss this with the team.

**Comment by i4ki** [comment]
As @soerenmartius mentioned, setting environment variables at the stack level is possible since `v0.8.0` release, then I will close this but feel free to reopen or create another issue if you have any problem with current design.
Thank you very much!
